fre/analysis/subtools.py

```diff
@@ -0,0 +1,103 @@
+from os import chdir, getcwd
+from pathlib import Path
+from subprocess import run
+from sys import executable
+from tempfile import TemporaryDirectory
+
+from analysis_scripts import find_plugins, run_plugin
+from yaml import safe_load
+
+
+def install(name, library_directory=None):
+    """Helper function to install possibly using a target directory.
+
+    Args:
+        name: String name of the package to install.
+        library_directory: Path to target directory you want to install the package in.
+    """
+    if library_directory:
+        run([executable, "-m", "pip", "install", f"--target={str(library_directory)}", name])
+    else:
+        run([executable, "-m", "pip", "install", name])
+
+
+def install_analysis_package(url, name=None, library_directory=None):
+    """Installs the analysis package.
+
+    Args:
+        url: URL to the github repository for the analysis package.
+        name: String name of the analysis-script package.
+        library_directory: Directory to install the package in.
+    """
+    if not url.startswith("https://"):
+        url = f"https://{url}"
+    if not url.endswith(".git"):
+        url = f"{url}.git"
+
+    if library_directory:
+        library_directory = Path(library_directory).resolve()
+
+    if name:
+        go_back_here = Path(getcwd())
+        with TemporaryDirectory() as tmpdirname:
+            tmp = Path(tmpdirname)
+            chdir(tmp)
+            run(["git", "clone", url, "scripts"])
+            chdir(tmp / "scripts" / "core" / "figure_tools")
+            install(".", library_directory)
+            chdir(tmp / "scripts" / "user-analysis-scripts" / name)
+            install(".", library_directory)
+            chdir(go_back_here)
+    else:
+        install(f"{url}@main", library_directory)
+
+
+def run_analysis(name, catalog, output_directory, output_yaml, experiment_yaml,
```

Comment:
so, if we didn't care about simultaneously having both `fre analysis FOO` calls and `from fre.freanalysis.subtools import FOO`, the solution would simply add a `@click.command()` to L54 above here.

But to preseve the import, define + dectorate another function with `@click.command()`, call it `_run_analysis`. all this function will do is talk to click, and pass arguments to `run_analysis`.

@singhd789 recently tidied all of this up [in `fre/pp`](https://github.com/NOAA-GFDL/fre-cli/blob/0ad7e5cb3dda25ab82d77c0c72624380819a15de/fre/pp/frepp.py) recently, making it a pretty clean one to follow.